packages/integration-test/src/setup.js

```diff
@@ -0,0 +1,174 @@
+        } catch (err) {
+            throw new AztecError(codes.NO_CONTRACT_ADDRESSES, {
+                message: 'No contract addresses exist for the given network ID in @aztec/contract-addresses',
+                networkId: this.networkId,
+            });
+        }
+    }
+
+    /**
+     * @method getContractPromises - get the Truffle contracts for desired contracts, with the provider set
+     */
+    getContractPromises() {
+        this.contractPromises = this.contractsToDeploy.map((nameOfContract) => this.getContractPromise(nameOfContract));
+    }
+
+    /**
+     * @method getContractPromise - get an object whose key is the name of a contract, and the value is a function that
+     * returns the Truffle representation of the contract
+     * @param {string} nameOfContract
+     */
+    getContractPromise(nameOfContract) {
+        const contractObject = {};
+        let extractedContractArtifact;
+        let extractedContractAddress;
+
+        try {
+            extractedContractArtifact = contractArtifacts[nameOfContract];
+        } catch (err) {
+            throw new AztecError(codes.NO_CONTRACT_ARTIFACT, {
+                message: 'Requested contract does not exist in @aztec/contract-artifacts',
+                requestedContract: nameOfContract,
+            });
+        }
+
+        try {
+            extractedContractAddress = this.contractAddresses[nameOfContract];
+        } catch (err) {
+            throw new AztecError(codes.NO_CONTRACT_ADDRESS, {
+                message: 'Requested contract does not have a deployed address in @aztec/contract-addresses',
+                requestedContract: nameOfContract,
+            });
+        }
+
+        contractObject[nameOfContract] = async () => {
+            const truffleRepresentation = this.getTruffleContractRepresentation(extractedContractArtifact);
+            return truffleRepresentation.at(extractedContractAddress);
+        };
+        return contractObject;
+    }
+
+    /**
+     * @method getTruffleContractRepresentation - get the Truffle contract representation of a contract, for a specific
+     * abi, bytecode and provider
+     * @param {Object} contractArtifact - truffle artifact for which the Truffle contract wrapper should be generated
+     */
+    getTruffleContractRepresentation(contractArtifact) {
+        const truffleRepresentation = TruffleContract({ abi: contractArtifact.abi, bytecode: contractArtifact.bytecode });
+        truffleRepresentation.setProvider(this.provider);
+        return truffleRepresentation;
+    }
+
+    /**
+     * @method getContracts - get the JavaScript objects representing contracts deployed at a specific address, for use
+     * in testing
+     * @returns {Object} Truffle contracts, representing the contract deployed at a specific address
+     */
+    async getContracts() {
+        const allContractObjects = (await Promise.all(
+            this.contractPromises.map(async (currentContractPromise, index) =>
```

Comment:
At this point `this.contractPromises` is an array of objects, each object having a contract name as a key, and a promise as a value. You currently don't make use of this data.

You can restructure `this.contractPromises` as an array of arrays, each inner array being `[contractName, promise]`, which would make both your map and your reduce simpler.

Reply:
Done